index.html

```diff
@@ -161,123 +144,134 @@ <h2>
-		with the mark's time value [[!HR-TIME-2]].
-    	</p>
-        <p>The <code>duration</code> attribute will return a <a href="https://w3c.github.io/hr-time/#idl-def-domhighrestimestamp"><code>DOMHighResTimeStamp</code></a>
-		of value 0.</p>
+        partial interface Performance {
+            void mark(DOMString markName);
+            void clearMarks(optional DOMString markName);
+            void measure(DOMString measureName, optional DOMString startMark, optional DOMString endMark);
+            void clearMeasures(optional DOMString measureName);
+        };
+      </pre>
+      <section>
+        <h2><dfn>mark()</dfn> method</h2>
+        <p>Stores a timestamp with the associated name (a "mark"). It MUST run these steps:</p>
+        <ol>
+          <li>Run the <a href="#ensure-name-is-valid">ensure name is valid</a> algorithm with <var>name</var> set to <var>markName</var>.</li>
+          <li>Create a new <a>PerformanceMark</a> object.</li>
+          <li>Set <code>name</code> to <var>markName</var>.</li>
+          <li>Set <code>entryType</code> to <code>DOMString "mark"</code>.</li>
+          <li>Set <code>startTime</code> to be the value that would be returned by the <code>Performance</code> object's <a data-cite="!HR-TIME-2#dom-performance-now"><code>now()</code></a> method.</li>
+          <li>Set <code>duration</code> to <code>0</code>.</li>
+          <li><a data-cite="!PERFORMANCE-TIMELINE-2#dfn-queue-a-performanceentry">Queue</a> the <a>PerformanceMark</a> object.</li>
+          <li id="stored_mark">Add the <code>PerformanceMark</code> object to the <a data-cite="!PERFORMANCE-TIMELINE-2#dfn-performance-entry-buffer">performance entry buffer</a>.</li>
+          <li>Return <strong>undefined</strong>.</li>
+        </ol>
+      </section>
+      <section>
+        <h2><dfn>clearMarks()</dfn> method</h2>
+        <p>Removes the stored timestamp with the associated name. It MUST run these steps:</p>
+        <ol>
+          <li>If <var>markName</var> is omitted, remove all <a>PerformanceMark</a> objects from the <a data-cite="!PERFORMANCE-TIMELINE-2#dfn-performance-entry-buffer">performance entry buffer</a>.</li>
+          <li>Otherwise, remove all <a>PerformanceMark</a> objects listed in the <a data-cite="!PERFORMANCE-TIMELINE-2#dfn-performance-entry-buffer">performance entry buffer</a> whose <var>name</var> matches<var>markName</var>.</li>
+          <li>Return <strong>undefined</strong>.</li>
+        </ol>
+      </section>
+      <section>
+        <h2><dfn>measure()</dfn> method</h2>
+        <p>Stores the <code>DOMHighResTimeStamp</code> duration between two marks along with the associated name (a "measure"). It MUST run these steps:</p>
+        <ol>
+          <li>Run the <a href="#ensure-name-is-valid">ensure name is valid</a> algorithm with <var>name</var> set to <var>measureName</var>.</li>

```

Comment:
Same comment as mark() on measureName.

Reply:
Sanity check.. the throw behavior for measure is not what we enforce in Chrome today. Did we agree that this is, indeed, the desired behavior?

Reply:
Filed https://github.com/w3c/user-timing/issues/36 to track this.

Reply:
> Same comment as mark() on measureName.

Done.

Reply:
Added 242a0ea to address #36